src/Http/Routing/src/Builder/RouteHandlerFilterExtensions.cs

```diff
@@ -29,9 +31,17 @@ public static RouteHandlerBuilder AddFilter(this RouteHandlerBuilder builder, IR
     /// <typeparam name="TFilterType">The type of the <see cref="IRouteHandlerFilter"/> to register.</typeparam>
     /// <param name="builder">The <see cref="RouteHandlerBuilder"/>.</param>
     /// <returns>A <see cref="RouteHandlerBuilder"/> that can be used to further customize the route handler.</returns>
-    public static RouteHandlerBuilder AddFilter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TFilterType>(this RouteHandlerBuilder builder) where TFilterType : IRouteHandlerFilter, new()
+    public static RouteHandlerBuilder AddFilter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TFilterType>(this RouteHandlerBuilder builder) where TFilterType : IRouteHandlerFilter
     {
-        builder.RouteHandlerFilters.Add(new TFilterType());
+        var type = typeof(TFilterType);
+        // We can provide the `MethodInfo` as an argument to the factory here so that `IRouteHandlerFilter`
+        // implementors can access the MethodInfo in their constructors.
+        var filterFactory = ActivatorUtilities.CreateFactory(type, Array.Empty<Type>());
```

Comment:
nit: 

```suggestion
        // We can provide the `MethodInfo` as an argument to the factory here so that `IRouteHandlerFilter`
        // implementors can access the MethodInfo in their constructors.
        var filterFactory = ActivatorUtilities.CreateFactory(typeof(TFilterType), Array.Empty<Type>());
```